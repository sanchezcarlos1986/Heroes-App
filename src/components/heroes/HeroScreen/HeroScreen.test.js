import React from 'react';
import {HeroScreen} from '~components';
import '@testing-library/jest-dom';
import {mount} from 'enzyme';
import {MemoryRouter, Route} from 'react-router-dom';

describe('HeroScreen', () => {
  let component;

  let historyMock;

  it('should renders redirect if !hero', () => {
    historyMock = {
      push: jest.fn(),
      goBack: jest.fn(),
      length: 10,
    };

    component = mount(
        <MemoryRouter>
          <HeroScreen history={historyMock} />
        </MemoryRouter>,
    );
    expect(component).not.toBeNull();
    expect(component.find('Redirect').exists()).toBe(true);
  });

  it('should renders hero screen and details', () => {
    component = mount(
        <MemoryRouter initialEntries={['/hero/dc-batman']}>
          <Route path="/hero/:heroId" component={HeroScreen} />
        </MemoryRouter>,
    );
    expect(component).not.toBeNull();
    expect(component).toMatchSnapshot();
    expect(component.find('img').prop('alt')).toBe('Batman');
  });

  it('should call PUSH', () => {
    historyMock = {
      push: jest.fn(),
      goBack: jest.fn(),
      length: 1,
    };

    component = mount(
        <MemoryRouter initialEntries={['/hero/dc-batman']}>
          <Route
            path="/hero/:heroId"
            component={() => <HeroScreen history={historyMock} />}
          />
        </MemoryRouter>,
    );

    component.find('button').simulate('click');
    expect(historyMock.push).toHaveBeenCalledWith('/');
    expect(historyMock.goBack).not.toHaveBeenCalled();
  });

  it('should call GO BACK', () => {
    historyMock = {
      push: jest.fn(),
      goBack: jest.fn(),
      length: 10,
    };

    component = mount(
        <MemoryRouter initialEntries={['/hero/dc-batman']}>
          <Route
            path="/hero/:heroId"
            component={() => <HeroScreen history={historyMock} />}
          />
        </MemoryRouter>,
    );

    component.find('button').simulate('click');
    expect(historyMock.push).not.toHaveBeenCalled();
    expect(historyMock.goBack).toHaveBeenCalled();
  });

  it('should return redirect if hero does not exists', () => {
    historyMock = {
      push: jest.fn(),
      goBack: jest.fn(),
      length: 10,
    };

    component = mount(
        <MemoryRouter initialEntries={['/hero/dc-batman438527590245']}>
          <Route
            path="/hero/:heroId"
            component={() => <HeroScreen history={historyMock} />}
          />
        </MemoryRouter>,
    );

    expect(component.html()).toBe('');
  });
});
