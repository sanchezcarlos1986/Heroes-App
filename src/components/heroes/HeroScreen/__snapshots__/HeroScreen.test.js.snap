// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`HeroScreen should renders hero screen and details 1`] = `
<div
  className="row mt-5"
>
  <div
    className="col-4"
  >
    <img
      alt="Batman"
      className="img-thumbnail animate__animated animate__fadeInLeft"
      src="../assets/heroes/dc-batman.jpg"
    />
  </div>
  <div
    className="col-8"
  >
    <h3>
      Batman
    </h3>
    <ul
      className="list-group list-group-flush"
    >
      <li
        className="list-group-item"
      >
        <strong>
          Alter ego: 
        </strong>
        Bruce Wayne
      </li>
      <li
        className="list-group-item"
      >
        <strong>
          Publisher: 
        </strong>
        DC Comics
      </li>
      <li
        className="list-group-item"
      >
        <strong>
          First Appareance: 
        </strong>
        Detective Comics #27
      </li>
    </ul>
    <h5>
      Characters
    </h5>
    <p>
      Bruce Wayne
    </p>
    <button
      className="btn btn-outline-info"
      onClick={[Function]}
    >
      Return
    </button>
  </div>
</div>
`;
